src/client/index.js

```diff
@@ -197,10 +197,12 @@ class Hammerhead {
             destLocation = parsedProxyUrl && parsedProxyUrl.destUrl;
         }
 
-        if (destLocation === 'about:blank' && isRelativeUrl(url))
+        if (isSpecialPage(destLocation) && isRelativeUrl(url))
             return;
 
-        this.win.location = urlUtils.getProxyUrl(url);
+        const proxyUrl = urlUtils.getProxyUrl(ensureHostEndedTrailingSlash(url));
```

Comment:
1)Rewrite in two lines of code:
```js
url = ensureHostEndedTrailingSlash(url);

const proxyUrl = urlUtils.getProxyUrl(url);
```
2)Host, port and protocol is named as 'origin'. See https://developer.mozilla.org/ru/docs/Web/API/Location.
So, let's rename function to `ensureOriginTrailingSlash(url)`.
Add this comment in function body
```js
If you request an url containing only port, host and protocol
then browser adds the trailing slash itself.
```

Reply:
Done. May be add window.location.href in comment?

    // NOTE: If you request an url containing only port, host and protocol
    // then browser adds the trailing slash to window.location.href.